### PR TITLE
[Relay][Convert Layout] Enable layout transformation for image.resize op

### PR DESCRIPTION
**Pull request by jtuyls**
This PR adds layout transformation support for the image.resize operation. This is useful for custom codegens as it allows creating larger partitions with the same layout in segmentation models (e.g. deeplabv3). In particular, this allows the Vitis AI codegen to offload a larger part of the model to the DPU accelerator if applicable.

@anijain2305 @yzhliu @trevor-m Could you please help with reviewing this PR?

### CODE REVIEW

File: python/tvm/relay/op/image/_image.py [hunk 1]
```diff
@@ -58,6 +58,38 @@ def compute_resize(attrs, inputs, out_type):
...
+        List of layouts defining our desired
+        layout for the data input.
+
+    Returns
+    -------
+    result : tvm.relay.Expr
+        The transformed expr
+    """
+    # pylint: disable=import-outside-toplevel
+    from tvm import relay
```
Comment (comaniac) [inline python/tvm/relay/op/image/_image.py:83]: Is there any reason for lazy import?

File: python/tvm/relay/op/image/_image.py [hunk 2]
```diff
@@ -58,6 +58,38 @@ def compute_resize(attrs, inputs, out_type):
...
+
+    new_attrs = dict(attrs)
+    assert len(desired_layouts) == 1, "Only one desired layout is expected"
+    (desired_layout,) = map(str, desired_layouts)
```
Comment (comaniac) [inline python/tvm/relay/op/image/_image.py:87]: It's better to use list comprehension instead of `map`.

**Review by comaniac — COMMENTED** [review]
Otherwise LGTM

File: python/tvm/relay/op/image/_image.py (hunk 1 above)
Comment (jtuyls) [reply python/tvm/relay/op/image/_image.py:83]: No, I just based this on other layout conversion functions and they seem to always lazy load inside the function: https://github.com/apache/tvm/blob/2c67d7131bc820481aaaca921f9e76f5709051b3/python/tvm/relay/op/nn/_nn.py#L240 Not sure why actually.

File: python/tvm/relay/op/image/_image.py (hunk 2 above)
Comment (jtuyls) [reply python/tvm/relay/op/image/_image.py]: Changed it to `desired_layout = str(desired_layouts[0])`

File: python/tvm/relay/op/image/_image.py (hunk 1 above)
Comment (comaniac) [reply python/tvm/relay/op/image/_image.py:83]: I see. I'll let @anijain2305 check if this is necessary.

**Review by comaniac — APPROVED** [review]
LGTM

**Review by trevor-m — APPROVED** [review]
LGTM

**Comment by anijain2305** [comment]
Nice, this should help TensorRT as well.

File: python/tvm/relay/op/image/_image.py (hunk 1 above)
Comment (anijain2305) [reply python/tvm/relay/op/image/_image.py:83]: I see. I might have copied it from some other place. @jtuyls Can you please check if you can move the import to the top (or just import the right module like `image` if the relay import fails).

Comment (jtuyls) [reply python/tvm/relay/op/image/_image.py:83]: @anijain2305 Yes, it works with the import at the top. We can also replace it with `from .image import resize` at the top.

**Comment by comaniac** [comment]
@jtuyls please fix the CI.

**Pull request merged**